src/modules/70-pipeline/components/PipelineSteps/Steps/BambooStep/BambooStepBase.tsx

```diff
@@ -0,0 +1,413 @@
+        />
+        {getMultiTypeFromValue(formik.values.spec.connectorRef) === MultiTypeInputType.RUNTIME && (
+          <ConnectorConfigureOptions
+            style={{ marginTop: 6 }}
+            value={formik.values.spec.connectorRef as string}
+            type="String"
+            variableName="spec.connectorRef"
+            showRequiredField={false}
+            showDefaultField={false}
+            onChange={value => formik.setFieldValue('spec.connectorRef', value)}
+            isReadonly={readonly}
+            connectorReferenceFieldProps={{
+              accountIdentifier: accountId,
+              projectIdentifier,
+              orgIdentifier,
+              type: Connectors.Bamboo,
+              label: getString('connectors.bamboo.bamboo'),
+              disabled: readonly,
+              gitScope: { repo: repoIdentifier || '', branch, getDefaultFromOtherRepo: true }
+            }}
+          />
+        )}
+      </div>
+
+      <div className={cx(stepCss.formGroup, stepCss.lg, css.jobDetails)}>
+        <FormInput.MultiTypeInput
+          label={getString('pipeline.bamboo.planName')}
+          name="spec.planName"
+          useValue
+          selectItems={planDetails}
+          placeholder={
+            connectorRefFixedValue && getMultiTypeFromValue(connectorRefFixedValue) === MultiTypeInputType.FIXED
+              ? loadingPlans
+                ? getString('common.loadingFieldOptions', {
+                    fieldName: getString('common.subscriptions.tabs.plans')
+                  })
+                : plansError?.message
+                ? plansError?.message
+                : getString('pipeline.planNamePlaceholder')
+              : getString('select')
+          }
+          multiTypeInputProps={{
+            onTypeChange: (type: MultiTypeInputType) => formik.setFieldValue('spec.planName', type),
+            expressions,
+            selectProps: {
+              allowCreatingNewItems: true,
+              addClearBtn: true,
+              items: planDetails,
+              loadingItems: loadingPlans,
+              itemRenderer: planPathItemRenderer,
+              noResults: (
+                <NoTagResults
+                  tagError={plansError}
+                  isServerlessDeploymentTypeSelected={false}
+                  defaultErrorText={getString('pipeline.bambooStep.noPlans')}
+                />
+              )
+            },
+            onFocus: (e: React.FocusEvent<HTMLInputElement>) => {
+              if (
+                e?.target?.type !== 'text' ||
+                (e?.target?.type === 'text' && e?.target?.placeholder === EXPRESSION_STRING)
+              ) {
+                return
+              }
+              refetchPlans()
+            },
+            allowableTypes
+          }}
+        />
+        {getMultiTypeFromValue(formik.values.spec.planName) === MultiTypeInputType.RUNTIME && (
+          <ConfigureOptions
```

Comment:
```suggestion
          < SelectConfigureOptions
```
To support allowed values. Check [https://harness.atlassian.net/wiki/spaces/PD/pages/21173404394/Allowed+Values+-+Pipeline+Team+-+Dev+changes](https://harness.atlassian.net/wiki/spaces/PD/pages/21173404394/Allowed+Values+-+Pipeline+Team+-+Dev+changes)